packages/wonder-blocks-data/src/util/result-from-cache-entry.js

```diff
@@ -15,24 +15,21 @@ export const resultFromCacheEntry = <TData: ValidData>(
     }
 
     const {data, error} = cacheEntry;
-
-    if (data != null) {
+    if (error != null) {
         return {
-            status: "success",
-            data,
+            status: "error",
+            error,
         };
     }
 
-    if (error == null) {
-        // We should never get here ever.
+    if (data != null) {
         return {
-            status: "error",
-            error: "Loaded result has invalid state where data and error are missing",
+            status: "success",
+            data,
         };
     }
 
     return {
-        status: "error",
-        error,
+        status: "aborted",
     };
```

Comment:
To support the GQL framework, we need to allow `null` data, and so we have the `aborted` status as a first-class citizen. Hopefully I can add fetch aborting to WB Data in a future update to make full use of this.

Reply:
cc: @kevinbarabash so that you're aware!